server/api/impl/proxy/bindings.go

```diff
@@ -33,33 +31,51 @@ func mergeBindings(bb1, bb2 []*apps.Binding) []*apps.Binding {
 	return out
 }
 
-func (p *Proxy) GetBindings(cc *apps.Context) ([]*apps.Binding, error) {
+func (p *Proxy) GetBindings(debugSessionToken apps.SessionToken, cc *apps.Context) ([]*apps.Binding, error) {
 	allApps := p.store.App().GetAll()
 
 	all := []*apps.Binding{}
 	for _, app := range allApps {
-		appID := app.Manifest.AppID
+		manifest, err := p.store.Manifest().Get(app.AppID)
+		if err != nil {
+			// TODO Log error (chance to flood the logs)
+			// p.mm.Log.Debug("Could not load manifest. Error: " + err.Error())
+			continue
+		}
+
+		appID := app.AppID
 		appCC := *cc
 		appCC.AppID = appID
 		appCC.BotAccessToken = app.BotAccessToken
 
-		up, err := p.upstreamForApp(app)
-		if err != nil {
-			return nil, errors.Wrapf(err, "failed to make upstream for %s", appID)
-		}
-
 		// TODO PERF: Add caching
 		// TODO PERF: Fan out the calls, wait for all to complete
-		bindingsCall := app.Manifest.Bindings
-		if bindingsCall == nil {
-			bindingsCall = apps.DefaultBindingsCall
+		bindingsCall := *apps.DefaultBindingsCall
+		if manifest.Bindings != nil {
+			bindingsCall = *manifest.Bindings
 		}
 		bindingsCall.Context = &appCC
 
-		bindings, err := upstream.GetBindings(up, bindingsCall)
+		resp := p.Call(debugSessionToken, &bindingsCall)
+
+		if resp == nil || resp.Type != apps.CallResponseTypeOK {
+			// TODO Log error (chance to flood the logs)
```

Comment:
1/5 log at Debug level?

Reply:
My fear is that any error here will fail on each binding get, and therefore we will have one line of error every time a user in the system changes channel, which will create lots of messages, even for debug level.

Reply:
ticket it? We need some feedback to the developer... `/apps debug-bindings` maaaybe good enough though.